main/src/ca/uwaterloo/flix/language/phase/unification/BoolFormula.scala

```diff
@@ -197,40 +202,35 @@ object BoolFormula {
     *
     * The map `m` must bind each free variable in `f` to a type variable.
     */
-  private def toBoolType(f: BoolFormula, m: Bimap[VarOrEff, Int], loc: SourceLocation): Type = f match {
+  private def toBoolType(f: BoolFormula, m: Bimap[IrreducibleEff, Int], loc: SourceLocation): Type = f match {
     case True => Type.True
     case False => Type.False
     case Var(x) => m.getBackward(x) match {
       case None => throw InternalCompilerException(s"Unexpected unbound variable: '$x'.", loc)
-      case Some(VarOrEff.Var(sym)) => Type.Var(sym, loc)
-      case Some(VarOrEff.Eff(sym)) => throw InternalCompilerException(s"Unexpected effect in Boolean type: '$sym'.'", loc)
-      case Some(VarOrEff.Assoc(sym, arg)) => throw InternalCompilerException(s"Unexpected associated effect in Boolean type: '$sym'.'", loc)
+      case Some(IrreducibleEff.Var(sym)) => Type.Var(sym, loc)
+      case Some(IrreducibleEff.Eff(sym)) => throw InternalCompilerException(s"Unexpected effect in Boolean type: '$sym'.'", loc)
+      case Some(IrreducibleEff.Assoc(sym, arg)) => throw InternalCompilerException(s"Unexpected associated effect in Boolean type: '$sym'.'", loc)
+      case Some(BoolFormula.IrreducibleEff.JvmToEff(t)) => t
     }
     case Not(f1) => Type.mkNot(toBoolType(f1, m, loc), loc)
     case And(t1, t2) => Type.mkAnd(toBoolType(t1, m, loc), toBoolType(t2, m, loc), loc)
     case Or(t1, t2) => Type.mkOr(toBoolType(t1, m, loc), toBoolType(t2, m, loc), loc)
   }
 
   /**
-    * Union of variable and effect types.
+    * An irreducible effect.
     */
-  sealed trait VarOrEff
+  sealed trait IrreducibleEff
+
+  object IrreducibleEff {
```

Comment:
IMO we should generally split renamings like this into a separate PR since they tend to distract from the real content.

Reply:
True.